### PR TITLE
[NSA-6856] Add notifications emails for end user sub-services requests

### PR DESCRIPTION
**Pull request by ipredoi**
Jira ticket: https://dfe-secureaccess.atlassian.net/browse/NSA-6856

### CODE REVIEW

File: lib/handlers/subServiceRequested/subServiceRequestToApprovers.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const organisationsClient = new OrganisatonsClient(config.notifications.organisations);
+    const directoriesClient = new DirectoriesClient(config.notifications.directories);
+
+    const isMultipleSubServicesReq = data.requestedSubServices.length > 1;
+    let subject = isMultipleSubServicesReq
+      ? 'A user has requested access to multiple sub-services'
+      : 'A user has requested access to a sub-service';
+
+    const { envName } = config.notifications;
+    if (envName !== 'pr') {
```
Comment (malikimran34) [inline lib/handlers/subServiceRequested/subServiceRequestToApprovers.js:19]: what is envName !== 'pr'? do we need a default value?

**Review by malikimran34 — CHANGES_REQUESTED** [review]
Looks good Ionut! few minor things related to DRY principle

File: lib/handlers/subServiceRequested/subServiceRequestToApprovers.js (hunk 1 above)
Comment (ipredoi) [reply lib/handlers/subServiceRequested/subServiceRequestToApprovers.js:19]: 'pr' is the env name we had in config files for Prod - this condition will add the env name in the email subject in the lower envs to differentiate lower envs requests from PROD requirements - no default value is needed

File: lib/handlers/subServiceRequestActioned/subServiceRequestApproved.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const { getEmailAdapter } = require('../../infrastructure/email');
+
+const jobType = 'sub_service_request_approved';
+
+const process = async (config, logger, data) => {
+  try {
+    const templateName = 'sub-service-request-approved';
+    const isMultipleSubServicesReq = data.requestedSubServices.length > 1;
+    const subject = isMultipleSubServicesReq
+      ? 'New sub-services added to your DfE Sign-in account'
```
[the remark this answers is not in the document]
Comment (ipredoi) [reply lib/handlers/subServiceRequestActioned/subServiceRequestApproved.js:10]: I chose this approach over concatenation for easy updates in the future when we may have much different email subjects. Happy to discuss in order to take the best decision for this

File: lib/infrastructure/templates/renderTemplates/sub-service-request-approved/email/text.ejs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Dear <%=firstName%> <%=lastName%>,
+
+<% if (isMultipleSubServicesReq) { %>
```
[the remark this answers is not in the document]
Comment (ipredoi) [reply lib/infrastructure/templates/renderTemplates/sub-service-request-approved/email/text.ejs:3]: the text looks repetitive, however the content is slightly different based on the number of sub-services requested. Introducing variables for each different paragraph will create confusion in reading and following the code - happy to discuss

**Review by malikimran34 — APPROVED** [review]

**Review by malikimran34 — APPROVED** [review]

**Pull request merged**